src/fides/api/ops/api/v1/endpoints/messaging_endpoints.py

```diff
@@ -165,6 +168,76 @@ def get_active_default_config(*, db: Session = Depends(deps.get_db)) -> Messagin
     return messaging_config
 
 
+# this needs to come before other `/default/{messaging_type}` routes so that `/status`
+# isn't picked up as a path param
+@router.get(
+    MESSAGING_STATUS,
+    dependencies=[Security(verify_oauth_client, scopes=[MESSAGING_READ])],
+    response_model=MessagingConfigStatusMessage,
+    responses={
+        HTTP_200_OK: {
+            "content": {
+                "application/json": {
+                    "example": {
+                        "config_status": "configured",
+                        "detail": "Active default messaging service of type MAILGUN is fully configured",
+                    }
+                }
+            }
+        }
+    },
+)
+def get_messaging_status(
+    *, db: Session = Depends(deps.get_db)
+) -> MessagingConfigStatusMessage:
+    """
+    Determines the status of the active default messaging config
+    """
+    logger.info("Determining active default messaging config status")
+
+    # confirm an active default messaging config is present
+    messaging_config = MessagingConfig.get_active_default(db)
+    if not messaging_config:
+        return MessagingConfigStatusMessage(
+            config_status=MessagingConfigStatus.not_configured,
+            detail="No active default messaging configuration found",
+        )
+
+    try:
+        details = messaging_config.details
+        MessagingConfigRequestBase.validate_details_schema(
+            messaging_config.service_type, details
+        )
+    except Exception:
+        return MessagingConfigStatusMessage(
+            config_status=MessagingConfigStatus.not_configured,
+            detail=f"Invalid or unpopulated details on {messaging_config.service_type.value} messaging configuration",  # type: ignore
+        )
+
+    # confirm secrets are present and pass validation
+    secrets = messaging_config.secrets
+    if not secrets:
+        return MessagingConfigStatusMessage(
+            config_status=MessagingConfigStatus.not_configured,
+            detail=f"No secrets found for {messaging_config.service_type.value} messaging configuration",  # type: ignore
+        )
+    try:
+        get_schema_for_secrets(
+            service_type=messaging_config.service_type,  # type: ignore
+            secrets=secrets,
+        )
+    except Exception as e:
```

Comment:
so interestingly when `update_config_secrets` calls `get_schema_for_secrets` it catches both `KeyError` and `ValueError` - but i agree that i don't immediately see how a `KeyError` can be raised. but i figured a general exception handler here is fine since it indicates _something_ is wrong with the secrets, and that's really all we're worried about with this endpoint.

but i know that it's generally not great practice to catch all types of exceptions broadly. i guess i feel like including the `str(e)` can give any additional info about what the specific exception is so that may mitigate things?

now that i think about it though, we should probably be logging that `str(e)` (and wrap it in our `Pii`) and _not_ return it in the endpoint, since it could potentially have sensitive data....

i'll take a closer look at reworking this a bit, thanks for pointing it out!

Reply:
Just an FYI, I'm fine with whatever decision you all make her. You can catch both `ValueError` and `KeyError` in the same catch: `except (ValueError, KeyError) as e:`

Reply:
updated in https://github.com/ethyca/fides/pull/2690/commits/f5b36f74bfe0251f1725f6c9b7cc1570e82b2f44, resolving for now but we can create a follow-up if it needs any adjustment